packages/playwright/src/runner/dispatcher.ts

```diff
@@ -326,6 +326,7 @@ class JobDispatcher {
     result.error = result.errors[0];
     result.status = params.status;
     result.annotations = params.annotations;
+    test.annotations = params.annotations; // last test result wins
```

Comment:
To make sure these are two different arrays:

```suggestion
    test.annotations = [...params.annotations]; // last test result wins
```